type-equality-compat/src-hetero/Data/Type/Equality/Hetero.hs

```diff
@@ -0,0 +1,68 @@
+{-# LANGUAGE CPP                 #-}
+#if __GLASGOW_HASKELL__ <800
+#error "Trying to compiled Data.Type.Equality.Hetero module with GHC <7.6"
```

Comment:
Why not just re-export `(:~~:)(..)` from `Data.Type.Equality` on newer GHCs?

Reply:
This module does export `:~~:` on newer GHCs? See line 23-25.

This error is a redundancy check to catch "I'm trying to compile this with GHC-7.x"

Reply:
Oops, you're absolutely right. I think I meant to comment on a different part of the PR.